plugins/ccgx/fu-ccgx-dmc-common.h

```diff
@@ -0,0 +1,310 @@
+	* 4 - 0xFF - Reserved
+	*/
+	guint8          model;
+} DmcDockIdentity;
+
+/* this structure defines the fields of status of a specific device  */
+typedef struct __attribute__((packed)) {
+	/* device ID of the device */
+	guint8          device_type;
+	/* component ID of the device */
+	guint8          component_id;
+	/* image mode of the device - single image/ dual symmetric/ dual
+	 * assymetric image > */
+	guint8          image_mode;
+	/* current running image */
+	guint8          current_image;
+	/* image status
+	 * b7:b4 => Image 2 status
+ 	 * b3:b0 => Image 1 status
+	 *  0 = Unknown
+	 *  1 = Valid
+	 *  2 = Invalid
+	 *  3-0xF = Reserved
+	 */
+	guint8          img_status;
+	/* padding */
+	guint8          reserved_0[3];
+	/* complete fw version 8 bytes for bootload, image1 and image2. 8 byte
+	* for fw version and application version */
+	guint8          fw_version[24];
+} DmcDevxStatus;
+
+/* this structure defines the fields of data returned when reading dock_status */
+typedef struct __attribute__((packed)) {
+	/* overall status of dock. see DmcDeviceStatus */
+	guint8          device_status;
+	/* eevice count */
+	guint8          device_count;
+	/* length of status bytes including dock_status,
+	devx_status for each device */
+	guint16         status_length;
+	/* dock composite version m_fwct_info */
+	guint32        composite_version;
+	/* fw status of device of interest */
+	DmcDevxStatus   devx_status[DMC_DOCK_MAX_DEV_COUNT];
+} DmcDockStatus;
+
+/* This structure defines the fields of data returned when reading an interrupt
+* from DMC */
+typedef struct __attribute__((packed)) {
+	guint8         opcode;
+	guint8         length;
+	guint8         data[DMC_INTERRUPT_DATA_LEN];
+} DmcIntRqt;
+
+/* this structure defines header structure of FWCT */
+typedef struct __attribute__((packed)) {
+	guint32         signature;
+	guint16         size;
+	guint8          checksum;
+	guint8          version;
+	guint8          custom_meta_type;
+	guint8          cdtt_version;
+	guint16         vid;
+	guint16         pid;
+	guint16         device_id;
+	guint8          reserv0[16];
+	guint32         composite_version;
+	guint8          image_count;
+	guint8          reserv1[3];
+} FwctInfo;
+
+typedef struct __attribute__((packed)) {
+	guint8          device_type;
+	guint8          img_type;
+	guint8          comp_id;
+	guint8          row_size;
+	guint8          reserv0[4];
+	guint32         fw_version;
+	guint32         app_version;
+	guint32         img_offset;
+	guint32         img_size;
+	guint8          img_digest[32];
+	guint8          num_img_segments;
+	guint8          reserv1[3];
+} FwctImageInfo;
+
+typedef struct __attribute__((packed)) {
+	guint8          img_id;
+	guint8          type;
+	guint16         start_row;
+	guint16         num_rows; /* size */
+	guint8          reserv0[2];
+} FwctSegmentationInfo;
+
+gchar   *fu_ccgx_dmc_version_to_string (guint32 val);
+gchar   *fu_ccgx_dmc_update_model_to_string (DmcUpdateModel val);
```

Comment:
copy the prototype style in existing files please.

Reply:
Replace it with fu_ccgx_dmc_update_model_type_to_string() on today commit (ccgx: fix code alignment, modify code for review)